plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergAnalyze.java

```diff
@@ -0,0 +1,570 @@
+                        + "('name',  null, null, 0, null, null, null),"
+                        + "(null,  null, null, null, 25, null, null)");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testDropStatsAccessControl()
+    {
+        String catalog = getSession().getCatalog().orElseThrow();
+        String schema = getSession().getSchema().orElseThrow();
+        String tableName = "test_deny_drop_stats";
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.sf1.nation", 25);
+
+        assertAccessDenied(
+                "ALTER TABLE " + tableName + " EXECUTE DROP_EXTENDED_STATS",
+                "Cannot execute table procedure DROP_EXTENDED_STATS on iceberg.tpch.test_deny_drop_stats",
+                privilege(format("%s.%s.%s.DROP_EXTENDED_STATS", catalog, schema, tableName), EXECUTE_TABLE_PROCEDURE));
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testReanalyze()
+    {
+        String tableName = "test_reanalyze";
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.sf1.nation", 25);
+        assertUpdate("ANALYZE " + tableName);
+        StatisticsFile initialStatsFile = Iterables.getOnlyElement(loadTable(tableName).currentSnapshot().statisticsFiles());
+        assertUpdate("ANALYZE " + tableName);
+        StatisticsFile newStatsFile = Iterables.getOnlyElement(loadTable(tableName).currentSnapshot().statisticsFiles());
+        assertNotEquals(initialStatsFile.location(), newStatsFile.location());
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testAnalyzeAndRollbackToSnapshot()
+    {
+        String schema = getSession().getSchema().orElseThrow();
+        String tableName = "test_analyze_and_rollback";
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.sf1.nation", 25);
+        long createSnapshot = getLatestSnapshotId(tableName);
+        assertUpdate("ANALYZE " + tableName);
+        long analyzeSnapshot = getLatestSnapshotId(tableName);
+        assertNotEquals(analyzeSnapshot, createSnapshot);
+
+        assertUpdate(format("CALL system.rollback_to_snapshot('%s', '%s', %s)", schema, tableName, createSnapshot));
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                "VALUES"
+                        // no NDV information after rollback_to_snapshot
+                        + "('nationkey', null, null, 0, null, '0', '24'),"
+                        + "('regionkey', null, null, 0, null, '0', '4'),"
+                        + "('comment', null, null, 0, null, null, null),"
+                        + "('name',  null, null, 0, null, null, null),"
+                        + "(null,  null, null, null, 25, null, null)");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testAnalyzeAndDeleteOrphanFiles()
```

Comment:
This currently fails.

Reply:
why is that? are stats files are cleaned?

Reply:
i don't remember, but maybe.